dali/base/dasess.cpp

```diff
@@ -151,6 +151,26 @@ static CriticalSection sessionCrit;
 #define CLDAPE_getpermtimeout (-1)
 #define CLDAPE_ldapfailure    (-2)
 
+#ifdef NULL_DALIUSER_STACKTRACE
+
+static void logNullUser(const char * func, int line = __LINE__)
+{
+    const double nullUserLogMinutes = 60;	//log NULL user requests no more than once this often
+    static time_t lastNullUserLogEntry = (time_t)NULL;
+    static CriticalSection nullUserLogCS;
```

Comment:
local static objects come with a hidden cost, c++ standard guarantees that they are initialized in a thread-safe way, meaning there's an implicit lock involved here.

To avoid the overhead, it is better to move these statics out of the function into module scope.